service/server.go

```diff
@@ -76,7 +80,7 @@ var (
 //
 // The health check server is a http.Server instance, which the port number is read from "config.Server.HealthzPort"
 // , and the handler is as follow - Handle HTTP GET request and always return HTTP Status OK (200) response.
-func NewServer(cfg config.Server, h http.Handler) Server {
+func NewServer(cfg config.Server, h http.Handler, dh http.Handler) Server {
```

Comment:
can you implement functional option params?

Reply:
@kevindiu 
something like this?
`func NewServer(cfg config.Server, serverOptions ...Option) Server {`